anacron/runjob.c

```diff
@@ -399,3 +411,134 @@ tend_children(void)
 	j++;
     }
 }
+
+/* Expand env variables in 'source' arg and save to 'result'*/
+void expand_envvar(const char *source, char *result, const int max_size) {
```

Comment:
Please replace const int with size_t. And similarly in other places where you use int for variables holding sizes.

The function should return int as success/failure. Whether then die-ing if the expansion fails at the caller is the right thing or not is another question. But it should be probably OK.

Reply:
I think it will be better to not die. The expansion will fail only if the result value is bigger than the max_size(254) which means that the configured email is invalid and the code will then use the not expanded value from env_get(). In that scenario, I think it's better to keep the jobs running and log the errors.